### PR TITLE
Keep session alive in background.

### PR DESCRIPTION
**Pull request by abhisekmazumdar**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ X ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Try creating a email and edit the template.
3. Walk away for at least 24 minutes.
4. Then try to save the changes of the email template.
5. You should be able to save it without losing any edited data.

### CODE REVIEW

File: app/bundles/CoreBundle/Assets/js/1.core.js [hunk 1]
```diff
@@ -72,6 +72,17 @@ mQuery( document ).ready(function() {
             e.preventDefault();
         }
     });
+
+    // Try to keep alive the session.
+    setInterval(function() {
+        mQuery.get('/s/keep-alive')
+          .done(function() {
+              console.log('Keep-alive tick');
```
Comment (abhisekmazumdar) [inline app/bundles/CoreBundle/Assets/js/1.core.js]: I will remove the debugging data after getting reviews and feedback.

**Comment by nick-vanpraet** [comment]
It's working, a bit too well. You'll have to multiple the gc_maxlifetime by 1000(right?) first, right now it's calling every 0.7 seconds.

It's also redirecting the call to the keepAlive to /s/login, I don't know enough about the symfony firewall system to help there though. But I imagine somewhere it's looking for a permission that probably does not exist.

File: app/bundles/CoreBundle/Assets/js/1.core.js [hunk 2]
```diff
@@ -72,6 +72,16 @@ mQuery( document ).ready(function() {
             e.preventDefault();
         }
     });
+
+    // Try to keep alive the session only if csrf-mautic_ajax_post cookie exists
+    if (mQuery.cookie('csrf-mautic_ajax_post')) {
```
Comment (abhisekmazumdar) [inline app/bundles/CoreBundle/Assets/js/1.core.js]: I found out that this snippet was being called when the user was not logged in too. I think adding a check for this by checking the CSRF cookies.

Comment (nick-vanpraet) [reply app/bundles/CoreBundle/Assets/js/1.core.js]: That cookie doesn't exist, only here does it exist: https://github.com/mautic/mautic/pull/13745

I think any secure page (with /s/) gets a session and csrf token regardless of login status. But I'm not sure any such pages are accessible to anonymous users besides /s/login. 

Basically, afaik, if that script gets loaded there will be a csrf token available. Be it a variable as in 5.x or a cookie as in the PR.

You could check for specifically /s/login as current path? Or leave it as is. Unless you found other paths where it is running where it shouldn't?

Comment (abhisekmazumdar) [reply app/bundles/CoreBundle/Assets/js/1.core.js]: I agree that looking for the CSRF token cookie was not the right approach. I found that it was indeed from the server.https://github.com/mautic/mautic/pull/13745.

Upon investigating further, I found that yes, the `/s/login` page is the only one with anonymous user access.
And adding proper checks around that helps to eliminate any updated call to `/s/keep-alive` route.

**Review by nick-vanpraet — APPROVED** [review]
Looks good, I tested it in gitpod by lowering the session lifetime and forcing garbage cleanup and I saw it calling the resource every half of the session lifetime. And it also works, I went to edit an email, opened the builder, left it alone for a while and when I came back I could save the template instead of getting a csrf error.

**Review by kuzmany — APPROVED** [review]
I did basic tests and works for me.
Code looks good too 👍 
Thanks

**Comment by kuzmany** [comment]
Even if the code looks good to me, I add a label for code review needed, to allow for a final check by @escopecz. Then we can proceed.

**Review by escopecz — APPROVED** [review]
Looks good to me. Thanks!

**Comment by escopecz** [comment]
@all-contributors please add @abhisekmazumdar for code

**Pull request merged**